### PR TITLE
Update v2 disperser protos

### PR DESCRIPTION
**Pull request by ian-shim**
## Why are these changes needed?
Updating blob header & cert schema 

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: api/proto/common/common.proto [hunk 1]
```diff
@@ -29,13 +29,20 @@ message BlobCommitment {
...
 // BlobCertificate is what gets attested by the network
 message BlobCertificate {
-	uint32 version = 1;
-	bytes blob_key = 2;
-	BlobCommitment blob_commitment = 3;
-	repeated uint32 quorum_numbers = 4;
+	bytes blob_key = 1;
+	BlobHeader blob_header = 2;
 	uint32 reference_block_number = 5;
+	repeated string relays = 4;
```
Comment (cody-littley) [inline api/proto/common/common.proto:45]: We need to decide what form the relay identifier will take. If relays are registered on chain then we can use a short identifier, e.g. a `uint32` or similar. Passing the relay URLs here will be convenient but could bloat the size of the certificate.

File: api/proto/common/common.proto [hunk 2]
```diff
@@ -29,13 +29,20 @@ message BlobCommitment {
...
+	bytes payment_header_hash = 4;
+}
+
 // BlobCertificate is what gets attested by the network
 message BlobCertificate {
-	uint32 version = 1;
-	bytes blob_key = 2;
-	BlobCommitment blob_commitment = 3;
-	repeated uint32 quorum_numbers = 4;
+	bytes blob_key = 1;
```
Comment (cody-littley) [inline api/proto/common/common.proto:42]: Is the blob key going to be equivalent to the blob hash? If "yes", perhaps if we should just call this field `blob_hash`.

Comment (ian-shim) [reply api/proto/common/common.proto:42]: `blob_key` is used throughout the system as a `hash(blob_header)`. What's the blob hash?

Comment (cody-littley) [reply api/proto/common/common.proto:42]: Makes sense, I think I was mistaken about the blob hash thing.

File: api/proto/common/common.proto (hunk 1 above)
Comment (ian-shim) [reply api/proto/common/common.proto]: updated

File: api/proto/common/v2/common.proto [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import "common/common.proto";
+option go_package = "github.com/Layr-Labs/eigenda/api/grpc/common/v2";
+
+
+message BlobHeader {
+	// Blob version
+	uint32 version = 1;
+	repeated uint32 quorum_numbers = 2;
+	common.BlobCommitment commitment = 3;
+	bytes payment_header_hash = 4;
```
Comment (mooselumph) [inline api/proto/common/v2/common.proto:12]: Should we just include the payment header now so as to minimize api changes in the future?

File: api/proto/common/v2/common.proto [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+	// signature over keccak hash of the blob_header that can be verified by blob_header.account_id
+	bytes signature = 5;
+}
+
+// BlobCertificate is what gets attested by the network
+message BlobCertificate {
+	bytes blob_key = 1;
```
Comment (mooselumph) [inline api/proto/common/v2/common.proto:20]: Is the blob key needed when the blob header is here?

**Review by mooselumph — APPROVED** [review]

**Pull request merged**